### ISSUE TITLE
Dynamically linking `libelf` - following up from CNCF Legal Committee for license exception request

### ISSUE THREAD

**Issue by leogr** [falcosecurity/evolution#359]
**Motivation**

While reviewing our[ license exception request for `libelf` (LGPL licensed)](https://github.com/cncf/foundation/issues/629), the CNCF Legal Committee noted that `libelf` is currently statically linked inside the Falco binary. Instead, they recommend dynamically linking to `libelf` so that the library remains separable from Falco at runtime. This allows us to comply with the CNCF license policy.

This issue will track all related items to accomplish their request.

```[tasklist]
### Tasks
- [ ] https://github.com/falcosecurity/libs/pull/1666
- [ ] https://github.com/falcosecurity/falco/pull/3048
- [ ] https://github.com/falcosecurity/falco/pull/3053
```

**Comment by LucaGuerra** [comment]
I have tested the dynamic linking patches merged in master with the following configurations:
- deb installed in ubuntu
- rpm installed in fedora
- container images:
    - falco
    - falco-driver-loader
    - falco-no-driver
    - falco-distroless

As expected, with DEB and RPM packages the libelf dependency is downloaded via the OS package manager. The container images are working properly except for an issue identified in the distroless/wolfi container. I already opened a PR for it, once it's merged I think we can close this issue.
